extension/js/common/core/common.ts

```diff
@@ -87,8 +87,11 @@ export class Str {
     if (email.indexOf(' ') !== -1) {
       return false;
     }
-    email = email.replace(/\:8001$/, ''); // for MOCK tests until https://github.com/FlowCrypt/flowcrypt-browser/issues/4631
-    return /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i.test(
+    email = email.replace(/\:8001$/, ''); // for MOCK tests, todo: remove from production
+    // todo: `email-addresses` parser used by OpenPGP.js consider top-level domains to be valid for emails e.g. address@domain
+    // should we allow it too (or use `email-addresses` package when manipulating user ids in keys?
+    // For now, I'm explicitly `localhost` domain, which is perfectly legal for testing
+    return /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|localhost|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i.test(
```

Comment:
I'm re-using `Str.parseEmail` method to re-extract email from userids verified by OpenPGP.js, as those verification methods only return userid string (and all valid userids are supposed to have email address embedded). 
OpenPGP.js is using (`email-addresses` NPM package)[`https://www.npmjs.com/package/email-addresses`] to extract email address from the userid.
So several questions/issues arise:
- Should we use `email-addresses` package or our own version for the re-extraction?
- If using own version, should we validate the email address? `email-addresses` validation is more lenient, as it allows a top level domain only, e.g. `address@domain`. As I understand, this type of validation isn't convenient for `compose` boxes, should we have a separate validation option when extracting emails from userids, like `VALIDATE-LENIENT"?

I had to explicitly add `localhost` as a valid domain into the regex

Reply:
@tomholub ?

Reply:
I think it's good as you've done it. It would indeed be odd to allow `hello@yup` as a valid email when composing. Similarly, having two ways to verify based on context is unnecessary complication. I think it doesn't necessarily need 100% alignment with OpenPGP.js. If there is an edge case when one check passes and the other doesn't, the result is still a failed action. Actually there's currently three places that validate email: this, OpenPGP.js, Gmail API. So it will never be unified. As long as nobody reports a problem, I'd leave it.

Reply:
I'd remove the todo comment anyway.

Reply:
As per above comment I'd remove the `todo` since I think it's ok as is

Reply:
Done